source/beamline/AMDetector.h

```diff
@@ -298,10 +298,17 @@ int outputSize = indexStart.totalPointsTo(indexEnd);
 	/// Returns a newly created action (possibly list of actions) to perfrom the detector cleanup
 	virtual AMAction3* createCleanupActions();
 
-	/// Returns a (list of) actions that will do all necessary steps to do dark current correction. Your subclass needs to implement this function. Dwell time argument is in seconds.
-	virtual AMAction3* createDarkCurrentCorrectionActions(double dwellTime);
-	/// Returns an action that sets the current detector value as the dark current correction value. Your subclass will probably call this in createDarkCurrentCorrectionActions.
-	virtual AMAction3* createSetAsDarkCurrentCorrectionAction();
+	/// Returns new action that initiates a dark current measurement, uses the provided dwell time. An AMDetector subclass that can do dark current correction would have to implement this.
+	virtual AMAction3* createDarkCurrentMeasurementAction(double secondsDwell) { Q_UNUSED(secondsDwell) return 0; }
```

Comment:
I think it is technically used elsewhere: implemented in CLSBasicScalerChannelDetector and called in BioXASMainXASScanActionController.